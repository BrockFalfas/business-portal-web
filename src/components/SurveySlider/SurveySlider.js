import React from 'react';
import PropTypes from 'prop-types';
import { Slider } from 'antd';

import './SurveySlider.css';

export class SurveySlider extends React.Component {
  static propTypes = {
    description: PropTypes.string,
    votesCount: PropTypes.number,
  };

  static defaultProps = {
    votesCount: 0,
  };

  render() {
    const { description, votesCount, ...sliderProps } = this.props;
    return (
      <div className="SurveySlider">
        <p className="SurveySlider-description">{description}</p>
        <Slider className="SurveySlider-slider" {...sliderProps} />
        <div className="SurveySlider-results">{votesCount} responses</div>
      </div>
    );
  }
}

export default SurveySlider;
