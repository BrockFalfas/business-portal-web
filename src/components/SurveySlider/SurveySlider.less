@import "../../styles/variables.less";

.SurveySlider {
  padding: 1rem 2rem;

  .SurveySlider-description {
    font-size: 1.2rem;
    line-height: 1.6rem;
    font-weight: bold;
    margin: 0.5rem 0 1rem;
  }

  .SurveySlider-slider {
    margin: 2rem 0 1rem;
  }

  .SurveySlider-results {
    text-align: center;
    font-size: 1rem;
    color: @color-grey;
  }

  .ant-slider-rail {
    background-image: linear-gradient(
      to right,
      #ea5a2e 0%,
      #ade534 50%,
      #48c955 100%
    );
  }
  .ant-slider-track {
    background-color: transparent !important;
  }
  .ant-slider-track,
  .ant-slider-rail {
    height: 0.4rem;
    border-radius: 0.4rem;
  }
  .ant-slider-handle {
    width: 1.5rem;
    height: 1.5rem;
    margin-left: -0.75rem;
    margin-top: -0.55rem;
    border-color: @color-blue;
  }
}
