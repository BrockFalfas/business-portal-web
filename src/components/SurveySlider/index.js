import SurveySlider from './SurveySlider';

export default SurveySlider;
