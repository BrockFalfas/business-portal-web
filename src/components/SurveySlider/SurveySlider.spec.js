import React from 'react';
import { shallow } from 'enzyme';

import { SurveySlider } from './SurveySlider';

const initSurveySlider = overrides => shallow(<SurveySlider {...overrides} />);

describe('component: <SurveySlider/>', () => {
  it('should render without crashes', () => {
    const wrapper = initSurveySlider();
    expect(wrapper).toBeTruthy();
  });

  it('should render description, votes count and slider', () => {
    const wrapper = initSurveySlider({
      description: 'fake description',
      votesCount: 33,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass all other props to the slider', () => {
    const sliderProps = { other1: 'value1', other2: 'value2' };
    const wrapper = initSurveySlider(sliderProps);
    expect(wrapper.find('Slider').props()).toMatchObject(sliderProps);
  });
});
