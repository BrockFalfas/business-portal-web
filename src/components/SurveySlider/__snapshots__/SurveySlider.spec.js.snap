// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <SurveySlider/> should render description, votes count and slider 1`] = `
<div
  className="SurveySlider"
>
  <p
    className="SurveySlider-description"
  >
    fake description
  </p>
  <Slider
    className="SurveySlider-slider"
  />
  <div
    className="SurveySlider-results"
  >
    33
     responses
  </div>
</div>
`;
